website/static/js/projectorganizer.js

```diff
@@ -1,220 +1,1023 @@
+    remote: {
+        url: '/api/v1/search/projects/?term=%QUERY&maxResults=20&includePublic=yes&includeContributed=no',
+        filter: function (projects) {
+            return $.map(projects, function (project) {
+                return {
+                    name: project.value,
+                    node_id: project.id,
+                    category: project.category
+                };
+            });
+        },
+        limit: 10
+    }
+});
+
+/**
+ * Bloodhound is a typeahead suggestion engine. Searches here for users projects
+ * @type {Bloodhound}
+ */
+projectOrganizer.myProjects = new Bloodhound({
+    datumTokenizer: function (d) {
+        return Bloodhound.tokenizers.whitespace(d.name);
+    },
+    queryTokenizer: Bloodhound.tokenizers.whitespace,
+    remote: {
+        url: '/api/v1/search/projects/?term=%QUERY&maxResults=20&includePublic=no&includeContributed=yes',
+        filter: function (projects) {
+            return $.map(projects, function (project) {
+                return {
+                    name: project.value,
+                    node_id: project.id,
+                    category: project.category
+                };
+            });
+        },
+        limit: 10
+    }
+});
+
+/**
+ * Edits the template for the column titles.
+ * Used here to make smart folder italicized
+ * @param {Object} item A Treebeard _item object for the row involved. Node information is inside item.data
+ * @this Treebeard.controller Check Treebeard API for methods available
+ * @private
+ */
+function _poTitleColumn(item) {
+    var css = item.data.isSmartFolder ? 'project-smart-folder smart-folder' : '';
+    return m('span', { 'class' : css }, item.data.name);
+}
+
+/**
+ * Links for going to project pages on the action column
+ * @param event Click event
+ * @param {Object} item A Treebeard _item object for the row involved. Node information is inside item.data
+ * @param {Object} col Column options
+ * @this Treebeard.controller Check Treebeard API for methods available
+ * @private
+ */
+function _gotoEvent(event, item, col) {
+    window.location = item.data.urls.fetch;
+}
+
+/**
+ * Watching for escape key press
+ * @param {String} nodeID Unique ID of the node
+ */
+function addFormKeyBindings(nodeID) {
+    $('#ptd-' + nodeID).keyup(function (e) {
+        if (e.which === 27) {
+            $('#ptd-' + nodeID).find('.cancel-button-' + nodeID).filter(':visible').click();
+            return false;
+        }
+    });
+}
+
+/**
+ * The project detail popup is populated based on the row that it was clicked from
+ * @param {Object} theItem Only the item.data portion of A Treebeard _item object for the row involved.
+ */
+function createProjectDetailHTMLFromTemplate(theItem) {
+    var detailTemplateSource,
+        detailTemplate,
+        detailTemplateContext,
+        displayHTML;
+    detailTemplateSource = $('#project-detail-template').html();
+    detailTemplate = Handlebars.compile(detailTemplateSource);
```

Comment:
Compiling this template only needs to happen once. Perhaps declare `detailTemplateSource`  and `detailTemplate` outside of this function.